File: src/main/scala/com/abaddon83/burraco/match/games/domainModels/burracoGames/completed/BurracoPoint.scala
package com.abaddon83.burraco.`match`.games.domainModels.burracoGames.completed

import com.abaddon83.burraco.`match`.games.domainModels.BurracoId
import com.abaddon83.burraco.`match`.games.domainModels.burracoGames.initialised.playerInGames.Burraco
import com.abaddon83.burraco.shares.decks.Card

case class BurracoPoint protected (
                         override protected val burracoId: BurracoId,
                         override protected val cards: List[Card]
                       ) extends Burraco {
}

object BurracoPoint{
  def apply(burraco: Burraco): BurracoPoint = {
    new BurracoPoint(burraco.getBurracoId(), burraco.showCards)
  }
}

File: src/main/kotlin/com/abaddon83/burracoGame/domainModels/burracoGameendeds/BurracoPoint.kt
package com.abaddon83.burracoGame.domainModels.burracoGameendeds

import com.abaddon83.burracoGame.shared.burracos.Burraco
import com.abaddon83.burracoGame.shared.burracos.BurracoIdentity
import com.abaddon83.burracoGame.shared.decks.Card

data class BurracoPoint private constructor(
        override val identity: BurracoIdentity,
        override val cards: List<Card>
): Burraco() {
    constructor(burraco: Burraco): this(burraco.identity(),burraco.showCards())

}